dao/src/main/java/org/thingsboard/server/dao/device/DeviceServiceImpl.java

```diff
@@ -324,6 +328,7 @@ public void deleteDevice(final TenantId tenantId, final DeviceId deviceId) {
             deviceCredentialsService.deleteDeviceCredentials(tenantId, deviceCredentials);
         }
         deleteEntityRelations(tenantId, deviceId);
+        entityStatisticsDao.deleteByEntityId(tenantId, deviceId);
```

Comment:
I am unsure that we should delete the statistics when we delete the device. We should delete it on the next billing cycle or use some trigger.